### ISSUE TITLE
I3042 rule when using !Sub function with S3 Bucket ARN

### ISSUE THREAD

**Issue by luizeboli** [aws-cloudformation/cfn-lint#2727]
### CloudFormation Lint Version

0.77.4

### What operating system are you using?

Mac

### Describe the bug

I'm trying to use `Fn::Sub` with a value imported from another Stack to populate `Resource` field of an IAM Role Policy Document

```yaml
Resource: 
  - !Sub
    - arn:aws:s3:::${SourceBucketS3}*
    - SourceBucketS3:
        Fn::ImportValue:
          !Join ['-', ['Base-Pipeline', !Ref Environment, 'SourceS3Bucket']]
```

This block produces the error `[cfn-lint] I3042: ARN in Resource PipelineRole contains hardcoded Partition in ARN or incorrectly placed Pseudo Parameters`

From [AWS docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-arn-format.html) this is the right format of an S3 arn.

### Expected behavior

No error should be shown

### Reproduction template

This template is working fine when creating a new Stack.

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  PipelineRole:
    Type: AWS::IAM::Role
    Properties:
      Path: /
      AssumeRolePolicyDocument: |
        {
          "Statement": [{
            "Effect": "Allow",
            "Principal": { "Service": ["codepipeline.amazonaws.com" ]},
            "Action": ["sts:AssumeRole"]
          }]
        }
      Policies:
        - 
          PolicyName: "root"
          PolicyDocument:
            Version: "2012-10-17"
            Statement:
              - 
                Effect: "Allow"
                Action:    
                  - s3:PutObject               
                Resource: 
                  - !Sub
                    - arn:aws:s3:::${SourceBucketS3}*
                    - SourceBucketS3:
                        Fn::ImportValue:
                          !Join ['-', ['Base-Pipeline', 'SourceS3Bucket']]
```

**Comment by kddejong** [comment]
`[cfn-lint] I3042: ARN in Resource PipelineRole contains hardcoded Partition in ARN or incorrectly placed Pseudo Parameters`

`I` based rules are just informational.  What this is referencing is the idea that to increase template re-use you should use: 
```yaml
Resource: 
  - !Sub
    - arn:${AWS::Partition}:s3:::${SourceBucketS3}*
    - SourceBucketS3:
        Fn::ImportValue: 
          !Join ['-', ['Base-Pipeline', 'SourceS3Bucket']]
```
